trie/slimtrie_scan.go

```diff
@@ -0,0 +1,426 @@
+package trie
+
+import (
+	"bytes"
+
+	"github.com/openacid/low/bitmap"
+)
+
+// NextRaw returns next key-value pair in []byte.
+type NextRaw func() ([]byte, []byte)
+
+// WalkFn is used when walking the slimtrie. Takes a
+// key and value, returning false if iteration should
+// be terminated.
+//
+// If withValue is false when calling NewIter(), value is always nil.
+//
+// Since 0.5.11
+type WalkFn func(key []byte, value []byte) bool
+
+// ScanFrom iterates key-values in the slimtrie and pass every key-value pair to
+// a callback function fn.
+//
+// The iteration starts from `start`, including the
+// starting key if includeStart is true.
+// The key and value it passes to fn are temporary slice []byte, i.e., next time calling
+// fn, the previously returned slice will be invalid.
+//
+// If withValue is false, the value passed to fn is always nil.
+//
+// ScanFrom requires a full slimtrie to work, i.e., created with NewSlimTrie(... Opt{Complete: Bool(true)}).
+//
+// Since 0.5.11
+func (st *SlimTrie) ScanFrom(
+	start string, includeStart bool,
+	withValue bool, fn WalkFn) {
+
+	startPath, startEqual := st.getGEPath(start)
+	nxt := st.newIter(startPath, startEqual && !includeStart, withValue)
+
+	for {
+		key, value := nxt()
+		if key == nil {
+			break
+		}
+
+		if !fn(key, value) {
+			break
+		}
+	}
+}
+
+// ScanFromTo is similar to ScanFrom except it accepts an additional ending boundary (end, includeEnd)
+//
+// Since 0.5.11
+func (st *SlimTrie) ScanFromTo(
```

Comment:
Curious if we need `includeStart` and `includeEnd` since iteration usually includes the start and excludes the end (e.g. in a slice `[start:end`).

Reply:
If you need to support pagination you may need an open start. E.g., aws-s3 responds 1000 items a time, e.g., `["abc", ... "rrr"]`.
Without an open start, the query for the next 1000 items has no way to accurately describe that it needs the next item after `"rrr"`.

A close end is rarely used either, in my experience. But an `includeStart` without an `includeEnd` looks weird. 😕 

I'll release `0.5.11` with this feature in one or two days then.